glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+    let mut ops = vec![SegmentOp::Retain; segments.len()];
+    for (ix, op) in todo {
+        // ix isn't always right because we didn't know the number of segments
+        // before, and we handle the first two points at the end
+        ops[ix % segments.len()] = op;
+    }
+
+    let mut els = Vec::with_capacity(path.elements().len());
+    let mut iter = segments
+        .iter()
+        .zip(ops)
+        .filter_map(|(seg, op)| op.apply(*seg))
+        .peekable();
+
+    // we will use `PathSeg::as_path_el` below, but that doesn't handle moveto
+    els.push(PathEl::MoveTo(iter.peek().unwrap().start()));
+    els.extend(iter.map(|seg| seg.as_path_el()));
+    if is_closed {
+        els.push(PathEl::ClosePath);
+    }
+    Some(BezPath::from_vec(els))
+}
+
+#[derive(Clone, Debug)]
+enum SegmentOp {
+    Retain,
+    Delete,
+    Replace(PathSeg),
+}
+
+impl SegmentOp {
+    fn apply(self, seg: PathSeg) -> Option<PathSeg> {
+        match self {
+            SegmentOp::Retain => Some(seg),
+            SegmentOp::Delete => None,
+            SegmentOp::Replace(seg) => Some(seg),
+        }
+    }
+}
+
+/// A potentially open corner.
+///
+/// This is any line segment, along with the preceding and following path segments.
+struct ErasureCandidate {
+    // index of the *first* participating segment, which contains prev_pt
+    ix: usize,
+    seg: Line,
+    prev: PathSeg,
+    next: PathSeg,
+}
+
+impl ErasureCandidate {
+    //https://github.com/googlefonts/glyphsLib/blob/74c63244fdb/Lib/glyphsLib/filters/eraseOpenCorners.py#L66-L71
+    fn points_are_right_of_line(&self) -> bool {
+        let prev_point = match self.prev {
+            PathSeg::Line(line) => line.start(),
+            PathSeg::Quad(quad_bez) => quad_bez.p1,
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p2,
+        };
+        let next_point = match self.next {
+            PathSeg::Line(line) => line.end(),
+            PathSeg::Quad(quad_bez) => quad_bez.p1,
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p1,
+        };
+
+        !(point_is_left_of_line(self.seg, prev_point)
+            || point_is_left_of_line(self.seg, next_point))
+    }
+
+    /// The intersection of the two path segments
+    ///
+    /// If this is an open corner, the intersection point will be the new corner.
+    fn intersection(&self) -> Option<Intersection> {
+        seg_seg_intersection(self.prev, self.next)
+    }
+}
+
+//https://github.com/googlefonts/glyphsLib/blob/74c63244fdbe/Lib/glyphsLib/filters/eraseOpenCorners.py#L14
+// 'left' from the perspective of an observer standing on line.p0 and lookign at line.p1?
+fn point_is_left_of_line(line: Line, point: Point) -> bool {
+    let Line { p0: a, p1: b } = line;
+    (b.x - a.x) * (point.y - a.y) - (b.y - a.y) * (point.x - a.x) >= 0.0
+}
+
+/// given a slice of path elements, determine which *segments* might be open corners.
+fn iter_erasure_candidates(
+    els: &[PathEl],
+    is_closed: bool,
```

Comment:
IIUC is_closed is a functions of els, I suggest computing it rather than passing it in